### PR TITLE
fix rockcraft.yaml for 1.6.1

### PR DESCRIPTION
**Pull request by Abuelodelanada**
This PR fixes rockcraft.yaml for `1.6.1`

# How to test it

Just run: `rockcraft pack` in `1.6.1` directory

### CODE REVIEW

File: 1.6.1/rockcraft.yaml [hunk 1]
```diff
@@ -28,6 +28,3 @@ parts:
       install -D -m755 pushgateway ${CRAFT_PART_INSTALL}/bin/pushgateway
     organize:
       pushgateway: bin/pushgateway
-  prometheus-pushgateway:
-    source-tag: v1.6.1
-    build-snaps: go/1.19/stable
```
Comment (sed-i) [inline 1.6.1/rockcraft.yaml:33]: Is this something CI added?

File: 1.6.1/rockcraft.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ parts:
     plugin: go
     source: https://github.com/prometheus/pushgateway
     source-type: git
-    source-tag: "v1.6.0"
+    source-tag: "v1.6.1"
     build-snaps:
       - go/1.21/stable
```
Comment (sed-i) [inline 1.6.1/rockcraft.yaml:23]: Just confirming it needs to be 1.21 and not 1.19.

**Review by sed-i — APPROVED** [review]

File: 1.6.1/rockcraft.yaml (hunk 1 above)
Comment (Abuelodelanada) [reply 1.6.1/rockcraft.yaml:33]: Yes, I filled and issue: https://github.com/canonical/observability/issues/103

File: 1.6.1/rockcraft.yaml (hunk 2 above)
Comment (Abuelodelanada) [reply 1.6.1/rockcraft.yaml:23]: I just used the latest stable version of go.
Now I have checked that [pushgateway uses 1.19](https://github.com/prometheus/pushgateway/blob/master/go.mod#L3)

**Pull request merged**